cronjob_scripts/checkout_worktree.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+import subprocess
+
+
+def checkout_worktree_for_arch(target_arch: str):
+    """
+    Checkout a git worktree for the given target_arch, in /tmp.
+
+    This is required for reading the exclude files.
+
+    This is lifted directly from the old trigger-package-build.sh script, and is only expected to
+    work on linux/macos with dash/bash.
+    """
+    worktree_path = f"/tmp/cargo-quickinstall-{target_arch}"
+    bash_script = f"""
+        set -eux
+
+        rm -rf {worktree_path}
```

Comment:
temp dir won't work here unless I pull the whole thing inside out and turn it into a context manager. For now, I'm just going to keep the logic how it was.

I don't think shutil buys us anything because these scripts are only ever run on unix machines. I'm assuming that if we have maintainers on windows, they all have access to WSL or similar?

In theory I could use `git ls-files $BRANCH` and `git show $BRANCH:$FILE` to avoid having any checkouts of the "trigger/*" branches at all. I don't think this is worth the bother though, because the builders do a checkout when writing failures, and the symmetry is quite nice.

Reply:
True, let's keep it for now